Sources/Data/DeviceSnapshot.swift

```diff
@@ -241,3 +213,186 @@ public struct DeviceSnapshot: Codable, Equatable {
         self.userInfo = userInfo
     }
 }
+
+// MARK: Push Token
+
+class DevicePushToken: NSObject, NSCoding, Codable {
```

Comment:
Suggested alternative naming strategy for these classes: Append `Snapshot` instead of prepending `Device`. I.e.

- DeviceSnapshot
- PushTokenSnapshot
- LocationSnapshot
- etc.

Instead of:

- DeviceSnapshot
- DevicePushToken
- DeviceLocation
- etc.

Just my opinion, but I feel the `Snapshot` approach better "groups" these objects together and better describes  their intended purpose.